ocaml/xapi/xapi_host.ml

```diff
@@ -596,10 +596,24 @@ let list_methods ~__context =
 let is_slave ~__context ~host = not (Pool_role.is_master ())
 
 let ask_host_if_it_is_a_slave ~__context ~host =
-  let local_fn = is_slave ~host in
-  Server_helpers.exec_with_subtask ~__context "host.ask_host_if_it_is_a_slave" (fun ~__context ->
-    (Message_forwarding.do_op_on_localsession_nolivecheck ~local_fn ~__context
-      ~host (fun session_id rpc -> Client.Client.Pool.is_slave rpc session_id host)))
+  let ask_and_warn_when_slow ~__context =
+    let local_fn = is_slave ~host in
+    let timeout = 10 in
+    let task_name = Context.get_task_id __context |> Ref.string_of in
+    let ip, uuid = Db.Host.get_address ~__context ~self:host, Db.Host.get_uuid ~__context ~self:host in
+    let rec log_host_slow_to_respond timeout' () =
```

Comment:
The `timeout'` is here to distinguish it from from `timeout` but I think it would be fine to use `timeout` and shadow the outer timeout as we don't need access to it.

Reply:
hmm, i suppose it is also preferable here because it ensures i won't accidentally use `timeout` where i might have actually wanted to use `timeout'`. 

I have always thought of shadowing as being bad. It would be nice to get other toolstackers' opinions on shadowing

Reply:
Shadowing is bad when you have 2 modules defining the same value, and you open both of them: when you open the 2nd module code using the 1st module would break (or worse call the function from the wrong module if their types actually match). In this case it is better to refer to the function with `Module.function` instead of open (the compiler does warn when open shadows things). If that is too long you can define a local module or function alias and use that.

For values inside a function I do not find shadowing particularly bad, as long as both values refer to semantically the same thing. In some cases it is even helpful, like in the example you point out: it prevents you from accessing the wrong `timeout` value by accident.
I would object to calling everything `let x = ` though, in that case shadowing would be a bad thing, because it'd be hard to see what the `x` actually means in each case.